src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/TypeHelpers.cs

```diff
@@ -612,5 +615,72 @@ internal static bool IsType(this Type type, string @namespace, string name)
             return string.Equals(type.Namespace, @namespace, StringComparison.Ordinal) &&
                 string.Equals(type.Name, name, StringComparison.Ordinal);
         }
+
+        internal static MemberInfo GetOriginalDefinition(this MemberInfo member)
+        {
+            var declaringType = member.DeclaringType;
+            var declaringTypeOriginalDefinition = declaringType.GetGenericTypeDefinitionOrSelf();
+            if (declaringType.Equals(declaringTypeOriginalDefinition))
+            {
```

Comment:
It would be simpler to test declaringType.IsGenericType() here. Consider avoiding GetGenericTypeDefinitionOrSelf() here and inlining that method in the one other place it's used.